src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -241,10 +241,11 @@ def register_extra_cli_argument(command, dest, **kwargs):
     _cli_extra_argument_registry[command][dest] = CliCommandArgument(dest, **kwargs)
 
 def cli_command(module_name, name, operation,
-                client_factory=None, transform=None, table_transformer=None, expose_no_wait=False):
+                client_factory=None, transform=None, table_transformer=None,
+                expose_no_wait=False, no_wait_param=None):
```

Comment:
Having both `expose_no_wait` and `no_wait_param` seems redundant. If no_wait_param is None then you implicitly haven't enabled it. If it is anything else, then you _have_ enabled it. However, I do like this way, of simply specifying the parameter name as it accomplishes the same thing with a little more flexibility. 

I would add a check here to verify that the param `no_wait_param` refers to is a boolean type and throw if it is not. Otherwise someone will try and use a string field or something and things will work in mysterious ways.